zerver/views/__init__.py

```diff
@@ -360,6 +369,14 @@ def create_homepage_form(request, user_info=None):
     # providing it.
     return HomepageForm(domain=request.session.get("domain"))
 
+def create_realm_creation_form(request, user_info=None):
+    if user_info:
+        return RealmCreationForm(user_info, domain=request.session.get("domain"))
+    # An empty fields dict is not treated the same way as not
+    # providing it.
+    return RealmCreationForm(domain=request.session.get("domain"))
```

Comment:
Hmm, why do you need this function?  It seems like there are 2 1-line cases, and it's called once for each, so it might be cleaner to inline this below.  

While you're modifying this, I think it'd be great to make it more clear from the arguments how the two cases differ.